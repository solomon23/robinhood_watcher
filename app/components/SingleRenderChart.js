/*
  ChartistGraph re-renders too frequently.  Our use case only needs a single render once we have data.
*/
import React, { Component } from 'react'
import ChartistGraph from 'react-chartist'

type Props = {
  data: object,
  options: object
}

export default class SingleRenderChart extends Component<Props> {
  props: Props

  shouldComponentUpdate(next) {
    const { data } = this.props
    return data === null && next.props.data
  }

  render() {
    const { data, options } = this.props

    return (<ChartistGraph data={data} options={options} type="Line" />)
  }
}
