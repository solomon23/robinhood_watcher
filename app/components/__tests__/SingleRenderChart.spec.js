import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import SingleRenderChart from '../SingleRenderChart'

Enzyme.configure({ adapter: new Adapter() })

describe('SingleRenderChart component', () => {
  it('should render', () => {
    const component = shallow(<SingleRenderChart data={{ id: 1 }} options={{ options: true }} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should not re-render', () => {
    const component = shallow(<SingleRenderChart data={{ id: 1 }} options={{ options: true }} />)
    expect(component.instance().shouldComponentUpdate({ data: { id: 2 } })).toBeFalsy()
  })
})
