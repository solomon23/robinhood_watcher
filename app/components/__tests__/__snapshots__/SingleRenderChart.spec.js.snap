// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleRenderChart component should render 1`] = `
<ChartistGraph
  data={
    Object {
      "id": 1,
    }
  }
  options={
    Object {
      "options": true,
    }
  }
  type="Line"
/>
`;
